### ISSUE TITLE
instance t4g not available

### ISSUE THREAD

**Issue by charles-lebrun** [terraform-aws-modules/terraform-aws-eks#1734]
## Description

I'm trying to add node group with t4g instance but got an AWS internal Error

I tried to create the node group with AWS CLI, and i dont got error
```
aws eks create-nodegroup --cluster-name my-cluster-name --nodegroup-name my-nodegroup-name --subnets subnet-0xxxxxxx --instance-types t4g.micro --ami-type AL2_ARM_64 --remote-access ec2SshKey=mySshKeyPair --node-role arn:aws:iam::xxxxxxxx:role/my-iam-name
```

## Versions
Terraform v1.1.2
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v3.70.0
+ provider registry.terraform.io/hashicorp/external v2.1.0

## Reproduction
workspace prod
apply terraform change
get error

````
Error: error waiting for EKS Node Group (EKS-Tiwas-prod:horse) to create: unexpected state 'CREATE_FAILED', wanted target 'ACTIVE'. last error: 1 error occurred:
│ 	* : InternalFailure: Amazon EKS was unable to process your request. Please try again.
````
### Code Snippet to Reproduce
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # aws_eks_node_group.horse is tainted, so must be replaced
-/+ resource "aws_eks_node_group" "project" {
      ~ arn                    = "arn:aws:eks:eu-west-3:XXXXXXX:nodegroup/EKS-prod/project/0abf0d83-8307-4314-b2c2-eefbb33a8da4" -> (known after apply)
      ~ disk_size              = 20 -> (known after apply)
      ~ id                     = "EKS-prod:project" -> (known after apply)
      ~ instance_types         = [
          - "t3a.medium",
          + "t4g.medium",
        ]
      + node_group_name_prefix = (known after apply)
      ~ release_version        = "1.5.1-ba2ba3e0" -> (known after apply)
      ~ resources              = [
          - {
              - autoscaling_groups              = []
              - remote_access_security_group_id = ""
            },
        ] -> (known after apply)
      ~ status                 = "CREATE_FAILED" -> (known after apply)
      - tags                   = {} -> null
      ~ tags_all               = {} -> (known after apply)
      ~ version                = "1.21" -> (known after apply)
        # (7 unchanged attributes hidden)


      ~ update_config {
          ~ max_unavailable            = 1 -> (known after apply)
          ~ max_unavailable_percentage = 0 -> (known after apply)
        }
        # (1 unchanged block hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

I tried several t4g instance type, but alway got the same error with terraform

**Comment by daroga0002** [comment]
you cannot mix t3a and t4g as those are different architectures (x86 vs arm64), what is more they require different AMIs

**Comment by charles-lebrun** [comment]
I have no issue when i change in my terraform t3a.medium to m6g.medium (also graviton), the problem occur only with t4g class

when y do a plan terraform prevent me as i will do a replace and not a change, for me the issue is not here.

Maybe I miss something ?

**Comment by daroga0002** [comment]
please run a debug in terraform as this seems be some AWS API related case

**Comment by charles-lebrun** [comment]
I run a full debug [terraform apply log](https://github.com/terraform-aws-modules/terraform-aws-eks/files/7822503/log.txt)

first step with destruction of node group work well,
creation fail, I found this error in EKS dashboard

```
AutoScalingGroupInvalidConfiguration | Amazon EC2 Autoscaling does not support the requested instanceType t4g.medium.
```

thanks for your help

**Comment by bryantbiggs** [comment]
@charles-lebrun are we ok to close this issue for now?

**Comment by charles-lebrun** [comment]
@bryantbiggs no, for me the error is still here, and do not come from aws

As I say first if a create the ressource with AWS CLI It's work, not with Terraform.

maybe terraform doesn't set the good AMI version ?

**Comment by bryantbiggs** [comment]
@charles-lebrun there are no restrictions imposed by the module and without seeing your configuration its hard to tell

**Comment by charles-lebrun** [comment]
this is the terraform code associate to my error

```
# Nodes EC2
resource "aws_eks_node_group" "horse" {
  cluster_name    = aws_eks_cluster.eks_cluster.name
  node_group_name = "horse"
  node_role_arn   = data.terraform_remote_state.iam.outputs.eks_node.arn
  subnet_ids      = data.terraform_remote_state.vpc.outputs.private_subnets

  instance_types = ["t4g.medium"]
  ami_type = "BOTTLEROCKET_ARM_64"
  capacity_type = "ON_DEMAND"


  labels = { "compute-type" : "ec2", "service" : "horse" }

  scaling_config {
    desired_size = 1
    max_size     = 2
    min_size     = 1
  }
}
```

I can see in EKS the nodegroup
but the ASG was not created

I also tried to edit the ASG with m6g.medium configuration (that work)
AWS console let me do the change
- edit launch configuration and change m6g.medium to t4g.medium
- go to ASG and change launch configuration version
- in ASG there is a instance restriction (launch configuration override) you have to remove it, if you try to change m6g.medium to t4g.medium here you got an error (don't know why)
- run a refresh instance in ASG and you got t4g.medium work well in EKS cluster (of course i got a warning on EKS because i manually edit the launch configuration: The Amazon EC2 Launch Template : lt-00b2618e424ebc47c has a new version (2) associated with your Autoscaling group, which is not managed by Amazon EKS. Expected Launch Template version : (1))

Is it more clear for you with that information ?

**Comment by charles-lebrun** [comment]
but it's true that if I go to my EKS cluster
clic and add node group, set the configuration

i got this error "AutoScalingGroupInvalidConfiguration | Amazon EC2 Autoscaling does not support the requested instanceType t4g.medium."

do you think it's an AWS error ?

**Comment by bryantbiggs** [comment]
example added here, you might be missing the `ami_type` https://github.com/terraform-aws-modules/terraform-aws-eks/blob/7babe877758ae0287975627a0135aa8b40c29e93/examples/eks_managed_node_group/main.tf#L139-L151

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.6 :tada:
